### DISCUSSION TITLE
Weird errors with "javax.lang.model.element.Modifier.SEALED"

### DISCUSSION THREAD

**Discussion by PlugDev-class** [Minestom/Minestom#1122] [Help] [answered]
Hey!  Everytime I try to start my server via IntelliJ, I'll get this error:
Could not find artifact org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-depchain:jar:3.1.4 in jitpack.io (https://jitpack.io)

So I tried to exclude the shrinkwrap with the following:
```
<exclusions>
  <exclusion>
    <groupId>org.jboss.shrinkwrap.resolver</groupId>
    <artifactId>shrinkwrap-resolver-depchain</artifactId>
  </exclusion>
</exclusions>
```

Now I get this error:
java: No enum constant javax.lang.model.element.Modifier.SEALED

My java is at the latest JDK-17 provided by oracle and my maven is up to date too.
But the weirdest part is, that I never used "javax.lang.model.element.Modifier.SEALED" this in my project..

Can someone help me please ^^

**Comment by VegetalDev** [comment, marked as answer]
Which version of Java is the Maven compiler plugin using?  It should look something like this:

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.10.1</version>
    <configuration>
        <release>17</release>
    </configuration>
</plugin>
```

Also if you haven't already, check the version of Java Maven is using by running `mvn -v` and make sure everything points to the correct JDK.

That being said, if you continue to have problems using Maven I would suggest switching to Gradle as your build management tool. Since Minestom uses it, you can take advantage of some really useful features like [composite builds](https://docs.gradle.org/current/userguide/composite_builds.html). You also won't have to exclude dependencies to prevent resolution failure. And writing custom tasks is much easier in Gradle than it is in Maven, if you ever find yourself in a situation where that'd be useful.
